src/utils/renderHtml/index.tsx

```diff
@@ -58,4 +61,59 @@ const renderHtml = (html: string = "", options: renderHtmlOptions = {}) => {
   });
 };
 
+/**
+ * `highlightHtml` adds highlights in the form of <mark> tags.
+ * @param html - The html string to parse
+ * @param searchTerm - The active search term as a regex expression
+ * @param highlights - The active highlights as a regex expression
+ * @returns - html string converted to an array of domnodes with highlighted text
+ */
+export const highlightHtml = (
```

Comment:
I would argue that the logic here is complicated enough to deserve its own folder/file.

Reply:
Added lots of file changes as I had to move some stuff around to make this work